src/components/trade/TradesTable/index.tsx

```diff
@@ -0,0 +1,176 @@
+    </tr>
+  )
+}
+
+export type Props = {
+  header?: JSX.Element
+  className?: string
+  numColumns?: number
+  data: Array<Order>
+}
+
+const TitleUid = styled(RowWithCopyButton)`
+  color: ${({ theme }): string => theme.grey};
+  font-size: ${({ theme }): string => theme.fontSizeDefault};
+  font-weight: ${({ theme }): string => theme.fontNormal};
+  display: flex;
+  align-items: center;
+`
+
+export const TradesTable = ({ header, className, numColumns, data }: Props): JSX.Element => {
+  const tableContext = React.useContext(TradesTableContext)
+  const [rows, setRows] = React.useState(<></>)
+
+  React.useEffect((): void => {
+    // for each entry in the data array invert the values of
+    // the limit price and execution price if toggled
+    const _rows = data.map((order, i): JSX.Element => {
+      interface price {
+        amount: BigNumber
+        fraction: TokenErc20
+      }
+      interface priceData {
+        denominator: price
+        numerator: price
+      }
+
+      const constructPrice = (data: priceData): string => {
+        const calculatedPrice = calculatePrice({
+          denominator: { amount: data.denominator.amount, decimals: data.denominator.fraction?.decimals },
+          numerator: { amount: data.numerator.amount, decimals: data.numerator.fraction?.decimals },
+        })
+
+        const displayPrice = (tableContext.isPriceInverted ? invertPrice(calculatedPrice) : calculatedPrice).toString(
+          10,
+        )
+        const formattedPrice = formatSmart({
+          amount: displayPrice,
+          precision: NO_ADJUSTMENT_NEEDED_PRECISION,
+          smallLimit: HIGH_PRECISION_SMALL_LIMIT,
+          decimals: HIGH_PRECISION_DECIMALS,
+        })
+        const buySymbol = order.buyToken ? safeTokenName(order.buyToken) : ''
+        const sellSymbol = order.sellToken ? safeTokenName(order.sellToken) : ''
+        const [baseSymbol, quoteSymbol] = tableContext.isPriceInverted
+          ? [sellSymbol, buySymbol]
+          : [buySymbol, sellSymbol]
+
+        return `${formattedPrice} ${quoteSymbol} for ${baseSymbol}`
+      }
+
+      return (
+        <tr key={i}>
+          <td>
+            <TitleUid textToCopy={order.uid} contentsToDisplay={order.shortId} />
+          </td>
+          <td className={i % 2 === 1 ? 'long' : 'short'}>{i % 2 === 1 ? 'Buy' : 'Sell'}</td>
+          <td>
+            {order.sellToken?.decimals.toFixed(3)} {order.sellToken?.symbol}
+          </td>
+          <td>
+            {order.buyToken?.decimals.toFixed(3)} {order.buyToken?.symbol}
+          </td>
+          <td>
+            {order.buyToken && order.sellToken
+              ? constructPrice({
+                  denominator: {
+                    amount: order.buyAmount,
+                    fraction: order.buyToken,
+                  },
+                  numerator: {
+                    amount: order.sellAmount,
+                    fraction: order.sellToken,
+                  },
```

Comment:
The order of denominator/numerator is inverted
![screenshot_2021-08-06_08-48-48](https://user-images.githubusercontent.com/43217/128537046-7f215f6f-41c7-4d59-add1-89290339a45f.png)